app/src/main/java/it/niedermann/nextcloud/deck/ui/helper/colorchooser/ColorChooser.java

```diff
@@ -43,6 +52,22 @@ public ColorChooser(Context context, @Nullable AttributeSet attrs) {
         inflater.inflate(R.layout.widget_color_chooser, this, true);
         ButterKnife.bind(this);
         initDefaultColors();
+
+        customColorPicker = (ColorPickerView) findViewById(R.id.customColorPicker);
+        brightnessSlideBar = findViewById(R.id.brightnessSlide);
+
+        customColorPicker.attachBrightnessSlider(brightnessSlideBar);
+
+        customColorPicker.setColorListener(new ColorEnvelopeListener() {
+            @Override
+            public void onColorSelected(ColorEnvelope envelope, boolean fromUser) {
+                String customColor =  "#" + envelope.getHexCode().substring(2);
```

Comment:
`envelope.getHexCode().substring(2);` is size of the string a contract made by the API? If not this should do `lenght` check.

Reply:
The color code that is returned by `getHexCode()` is 8 characters long, starting with the 2 characters for the transparency, followed by the regular 6 characters for colors (2 for each color). This line of code formats the return from `getHexCode()` so that it matches what the API is looking for (# + 6 char color code, ex: #F2BCCC).
I borrowed this approach from something I already found in the existing code, in EditBoardDialogFragment.java
`
this.fullBoard.board.setColor(colorChooser.getSelectedColor().substring(1));
`
and
`
this.colorChooser.selectColor("#" + fullBoard.getBoard().getColor());
`

Reply:
I think, that's just fine. We can simply expect it.